### ISSUE TITLE
[Power Up] Tackling quantum phase transitions and barren plateaus in VQE with tensor networks

### ISSUE THREAD

**Issue by ryanlevy** [XanaduAI/QHack2021#28]
### Team Name:

Hooked on Photonics

### Project Description:

One of the most promising applications of variational quantum algorithms is the study of condensed matter phenomena, such as quantum phase transitions, with near-term quantum computers. Yet a major challenge in the successful application of variational quantum algorithms is the barren plateau phenomenon, where gradients become exponentially small in the number of qubits [1]. While barren plateaus have been observed for certain types of variational quantum circuits and cost functions [2], it is unclear whether the phenomenon would significantly hinder the simulation of condensed matter systems. Our goal in this project is to explore this problem and develop strategies for avoiding barren plateaus in the study of quantum phase transitions.

In particular, we use the variational quantum eigensolver (VQE) [3] to find the ground states of the transverse field Ising model, a spin chain whose ground state is known to undergo a quantum phase transition. To avoid the barren plateau phenomenon in our analysis of this model, we train variational circuits with physically relevant structure, such as tree tensor networks (TTN) and the multi-scale entanglement renormalization ansatz (MERA) [4]. The MERA, a tensor network used to study quantum critical systems, is particularly well-suited to our task. We show that for our problem TTN’s and MERA’s generally produce larger gradients than a hardware-efficient ansatz (HEA) typically used in VQE and thereby are easier to train and help alleviate barren plateaus.

[1] McClean, J.R., Boixo, S., Smelyanskiy, V.N. et al. Barren plateaus in quantum neural network training landscapes. Nat Commun 9, 4812 (2018).
[2] M. Cerezo, Akira Sone, Tyler Volkoff, Lukasz Cincio, Patrick J. Coles. Cost-Function-Dependent Barren Plateaus in Shallow Quantum Neural Networks. arXiV:2001.00550 (2020).
[3] Peruzzo, A., McClean, J., Shadbolt, P. et al. A variational eigenvalue solver on a photonic quantum processor. Nat Commun 5, 4213 (2014).
[4] G. Evenbly and G. Vidal. Algorithms for entanglement renormalization. Phys. Rev. B 79, 144108 (2009).

### Source code:

https://github.com/echertkov/qhack_vqe_ttn

### Resource Estimate:

In our project, we are performing VQE optimizations over a significant space of variables. We are considering three different variational ansatz (TTN, MERA, HEA), many different Hamiltonian parameters, and many different numbers of qubits. For reference, a single 16-qubit VQE optimization step of a MERA ansatz takes us about 40 seconds (or 2.8 hours for full training) and we need to explore about 100 different parameter values, so this altogether takes about 280 hours. However, we would like to perform additional simulations with more qubits to clearly show how quantities such as the variance of cost-function gradients scale with qubit number. Measurement of a TTN gradient at 18 qubits takes around 4.5 minutes on Amazon Braket’s SV1 simulator per sample, so to fully explore multiple ansatz and parameters we estimate that we need around 30 hours of compute time, which would cost 0.075*1800=$135. Since the computational cost increases exponentially with the number of qubits, to perform simulations with more than 18 qubits we would need significant additional computational resources. Additional AWS credits will allow us to utilize Braket’s simulators and perform these large simulations.

We would also like to use our AWS credits to perform calculations on the IonQ QPU. In particular, we intend to observe the quantum phase transition in the TFI model by measuring observables such as <X> and <Z> magnetizations on our trained ansatze. The cost of this for a single ansatz would be (8 magnetic field values) * (2 observables) * ($0.30000+$0.01*2000 shots)= $324.8 / ansatz and we would like to consider two ansatz, the TTN and the MERA. 

Time permitting we could additionally study if our results extend to other models, such as the transverse plus longitudinal field Ising model, which is not classically solvable. This should incur similar costs and time, potentially doubling our estimated AWS expenses.

Total estimate:
```
280 Training the ansatze (large instances)
270 Measuring gradient variances for the ansatze (estimate for larger qubits)
650 Measuring observables on IonQ’s QPU
===
$1,200.00 ($2,400 with second Hamiltonian)
```

**Comment by co9olguy** [comment]
Thanks for the detailed draft submission @ryanlevy! :rocket:

**Comment by co9olguy** [comment]
Thanks for your Power Up Submission @ryanlevy !

To help us keep track of final submissions, we will be closing all of the [Power Up] issues. We ask you to open a new issue for your final submission. Please use [this pre-formatted [Entry] Issue template](https://github.com/XanaduAI/QHack/issues/new?assignees=&labels=submission&template=open-hackathon-final-submission.md&title=%5BENTRY%5D+Your+Project+Title). Note that for the final submission, the Resource Estimate requirement is replaced by a Presentation item.
